doc/sphinx/source/howto/tune-for-deployment.rst

```diff
@@ -0,0 +1,171 @@
+adjustments can increase performance. Likewise, a minimally-designed framework
+could offer significant performance-improvement opportunities by lowering overhead.
+
+For this reason, we're providing some of the more commonly-used options for 
+fine-tuning various kinds code deployments to the nGraph-enabled devices we 
+currently support. Watch this section for new updates. 
+
+
+Training Deep Neural Networks
+==============================
+
+This section details some of the *configuration options* and *environment variables* 
+that can be used to tune for optimal performance when your system already has a
+version of nGraph installed.  
+
+Before tweaking various environment variables, be aware that how the computation 
+gets deployed depends upon the data format that the model is using. NHWC or NCHW
+are the two more common layouts, and the ultimate runtime can vary greatly -- 
+even when all other factors are the same -- when this detail is overlooked.
+
+* **N** -- Number of images per batch
+* **H** -- Height of the image
+* **W** -- Width of the image
+* **C** -- Channel of the image (expressed as a number like 3 for RGB and 1 
+  for grayscale)
+
+MKL-DNN
+-------
+
+The following `KMP options`_ were originally optimized for `MKLDNN`_ projects 
+running models with the ``NCHW`` data layout; however, other configurations can 
+be explored. MKL-DNN is automatically enabled as part of an nGraph build; you do 
+*not* need to add MKL-DNN separately on the framework side to use these 
+configuration settings.  
+
+* ``KMP_BLOCKTIME`` Sets the time, in milliseconds, that a thread should wait 
+  after completing the execution of a parallel region, before sleeping.
+* ``KMP_AFFINITY`` Enables the runtime library to bind threads to physical 
+  processing units. 
+* ``KMP_SETTINGS`` Enables (``true``) or disables (``false``) the printing of 
+  OpenMP* runtime library environment variables during program execution.
+* ``OMP_NUM_THREADS`` Specifies the number of threads to use.
+
+
+Code deployment to nGraph-enabled Intel® Xeon®
+==============================================
+
+The list below includes recommendations on :term:`data layouts`, parameters, and 
+application configuration to achieve best performance running DNN workloads on 
+Intel® Xeon® (CPU processor) systems.
+
+Threading 
+---------
+
+The number of threads set by ``OMP_NUM_THREADS`` ought not exceed the number of 
+physical cores. The threads should be pinned to their respective physical cores 
+and actived as follows:
+
+* When ``HT=off``, ``KMP_AFFINITY=compact,granularity=fine``
+
+* When ``HT=on``, ``KMP_AFFINITY=compact,1,0,granularity=fine``
+
+
+Memory allocation 
+-----------------
+
+Buffer pointers should be aligned at the 64-byte boundary. NUMA policy should be 
+configured for local memory allocation (``numactl --localloc``)
+
+Convolution shapes
+^^^^^^^^^^^^^^^^^^
+* When **running inference, or training for forward-propagation and weight 
+  updates**,
+  
```

Comment:
When running inference....and weight updates, for best performance,

Reply:
Added